pkg/tsdb/cloudwatch/get_dimension_values_test.go

```diff
@@ -0,0 +1,56 @@
+package cloudwatch
+
+import (
+	"context"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudwatch"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/mocks"
+	"github.com/grafana/grafana/pkg/tsdb/cloudwatch/models"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetDimensionValues(t *testing.T) {
+	api := &mocks.MetricsAPI{Metrics: []*cloudwatch.Metric{
+		{MetricName: aws.String("Test_MetricName1"), Dimensions: []*cloudwatch.Dimension{{Name: aws.String("Test_DimensionName1"), Value: aws.String("Value1")}, {Name: aws.String("Test_DimensionName2"), Value: aws.String("Value2")}}},
+		{MetricName: aws.String("Test_MetricName2"), Dimensions: []*cloudwatch.Dimension{{Name: aws.String("Test_DimensionName1"), Value: aws.String("Value3")}}},
+		{MetricName: aws.String("Test_MetricName3"), Dimensions: []*cloudwatch.Dimension{{Name: aws.String("Test_DimensionName1"), Value: aws.String("Value4")}}},
+		{MetricName: aws.String("Test_MetricName4"), Dimensions: []*cloudwatch.Dimension{{Name: aws.String("Test_DimensionName1"), Value: aws.String("Value2")}}},
+	}}
+	executor := &cloudWatchExecutor{}
+
+	t.Run("Should not change non-wildcard dimension value", func(t *testing.T) {
+		query := getBaseQuery()
+		query.MetricName = "Test_MetricName1"
+		query.Dimensions = map[string][]string{"Test_DimensionName1": {"Value1"}}
+		queries, err := executor.getDimensionValues(context.Background(), api, "us-east-1", []*models.CloudWatchQuery{query})
```

Comment:
We can explicitly say the call is not made in these tests.

```suggestion
		queries, err := executor.getDimensionValues(context.Background(), nil, "us-east-1", []*models.CloudWatchQuery{query})
```